soda_machine.rb

```diff
@@ -7,12 +7,23 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    @sodas.find {|soda| soda.brand == soda_brand}
   end
 
   def sell(soda_brand)
+    #using the find method since it's already written and it returns the soda
+    sold_soda = find_soda(soda_brand)
+      if sold_soda
+        #cash will increase by the cost of the soda
+        @cash += sold_soda.price
+        @sodas.delete(sold_soda)
+      else
```

Comment:
The else is not really necessary -- by default you'll return nil if you don't specify something else to return.  Also, lines 20-26 should be unindented by one tab.